File: 1/03.pl
#!/usr/bin/perl
use strict;
use warnings;
my $s = "Now I need a drink, alcoholic of course, after the heavy lectures involving quantum mechanics.";
my @a = split(/\s+/, $s);
@a = map { s/\w+//; length($&) } @a;
print "@a\n";

File: 1/03.exs
s = "Now I need a drink, alcoholic of course, after the heavy lectures involving quantum mechanics."
Regex.split(~r/\s+/, s)
|> Enum.map(&(Regex.replace(~r/\W+/, &1, "")))
|> Enum.map(&String.length(&1))
|> IO.inspect
